src/components/action-popover/action-popover.test.tsx

```diff
@@ -0,0 +1,2303 @@
+      <ActionPopoverItem icon="email">Email Invoice</ActionPopoverItem>
+      <ActionPopoverItem icon="print" disabled>
+        Print Invoice
+      </ActionPopoverItem>
+    </ActionPopover>
+  );
+
+  await user.click(screen.getByRole("button"));
+  await user.click(screen.getByText("Print Invoice"));
+
+  expect(screen.getByRole("button", { name: "actions" })).not.toBeFocused();
+});
+
+test("pressing enter on a disabled menu item does not focus the menu button", async () => {
+  const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+
+  render(
+    <ActionPopover>
+      <ActionPopoverItem icon="email">Email Invoice</ActionPopoverItem>
+      <ActionPopoverItem icon="print" disabled>
+        Print Invoice
+      </ActionPopoverItem>
+    </ActionPopover>
+  );
+
+  await user.click(screen.getByRole("button"));
+  await screen.getByRole("button", { name: "Print Invoice" }).focus();
+  await user.keyboard("{Enter}");
+
+  expect(screen.getByRole("button", { name: "actions" })).not.toBeFocused();
+});
+
+test("clicking the menu button calls the onOpen prop", async () => {
+  const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+
+  const onOpen = jest.fn();
+
+  render(
+    <ActionPopover onOpen={onOpen}>
+      <ActionPopoverItem>example item</ActionPopoverItem>
+    </ActionPopover>
+  );
+
+  await user.click(screen.getByRole("button"));
+
+  expect(onOpen).toHaveBeenCalledTimes(1);
+});
+
+test("clicking the menu button stops any further event propagation", async () => {
+  const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+
+  const parentClickHandler = jest.fn();
+
+  render(
+    // eslint-disable-next-line jsx-a11y/click-events-have-key-events, jsx-a11y/no-static-element-interactions
+    <div onClick={parentClickHandler}>
+      <ActionPopover>
+        <ActionPopoverItem>example item</ActionPopoverItem>
+      </ActionPopover>
+    </div>
+  );
+
+  await user.click(screen.getByRole("button"));
+
+  expect(parentClickHandler).not.toHaveBeenCalled();
+});
+
+test("clicking the menu button focuses the first focusable element", async () => {
+  const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+
+  render(
+    <ActionPopover>
+      <ActionPopoverItem disabled>shouldn't be focused</ActionPopoverItem>
+      <ActionPopoverItem>example item</ActionPopoverItem>
+    </ActionPopover>
+  );
+
+  await user.click(screen.getByRole("button"));
+
+  expect(screen.getByRole("button", { name: "example item" })).toBeFocused();
+});
+
+test("clicking the menu button with the menu open closes the menu and calls the onClose function", async () => {
+  const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+
+  const onClose = jest.fn();
+
+  render(
+    <ActionPopover onClose={onClose}>
+      <ActionPopoverItem>example item</ActionPopoverItem>
+    </ActionPopover>
+  );
+
+  await user.click(screen.getByRole("button"));
```

Comment:
comment (non-blocking): I assume this and the button located below are the same element so we could store this in a variable etc

Reply:
yeah probably. There were a few tests where I can't do that because the `getByRole` call does a new query each time, so if the DOM has changed then it's needed (eg to assert the element is there to start with and then not there after an action), but looking at this one I think you're right that it should be OK to store the resulting element in a variable and reuse it.